tests/e2e/framework/kubernetes.go

```diff
@@ -898,6 +904,31 @@ spec:
 `
 )
 
+// Wait for Kubernetes to become active within kubernetesResponiveTimeout period
+// Does a simple kubectl get pods operation to determine if kubeapi is active
+// (TODO) sdake: This may be insufficient for a complete readiness check of Kubernetes
+func (k *KubeInfo) waitForKubernetes() error {
+	log.Info("Waiting for Kubernetes to become responsive")
+	cmd := fmt.Sprintf("kubectl get pods --kubeconfig=%s", k.KubeConfig)
+	timeout := time.Now().Add(kubernetesResponsiveTimeout)
+	for {
```

Comment:
nit: use the `pkg/test/util/retry` package instead of this loop.

Reply:
will take a look.  I copied this logic from the function right below that waits for some other part of k8s to be ready.

Reply:
thanks for the suggestion.  Code looks much cleaner now.  I am not sure I am reporting an error string correctly.  Can you make a suggestion?